torch/lib/c10d/Utils.cpp

```diff
@@ -231,6 +236,23 @@ int connect(
         if (!wait || (!anyRefused && !anyReset)) {
           throw;
         }
+
+        /*
+        if a timeout is specified, check time elapsed to see if we need to
+        timeout. A timeout is specified if timeout != kNoTimeout.
+        */
+        if (timeout != kNoTimeout) {
+          auto now =
+              std::chrono::high_resolution_clock::now().time_since_epoch();
+          auto nowMilliseconds =
+              std::chrono::duration_cast<std::chrono::milliseconds>(now)
+                  .count();
+
+          auto elapsedMilliseconds = nowMilliseconds - startMilliseconds;
+          if (elapsedMilliseconds > timeout.count()) {
+            throw std::runtime_error("Connecting to TCP store timed out.");
+          }
+        }
```

Comment:
No need to take `time_since_epoch`. You can use the `std::chrono::time_point` and `std::chrono::duration` directly without bothering with the unit (the duration class takes care of the comparison correctly if the unit is different):

```c++
const auto start = std::chrono::high_resolution_clock::now();

// ...

const auto elapsed = std::chrono::high_resolution_clock::now() - start;
if (elapsed > timeout) {
  // ...
}
```